documentation/how-to/prepare-for-production.md

```diff
@@ -0,0 +1,193 @@
+openssl req -new -key server.key -out server.csr -subj "/CN=<YOUR_HOST_NAME>"
+
+# Create self-signed certificate
+openssl x509 -req -days 365 -in server.csr -signkey server.key -out server.crt -extfile <(printf "subjectAltName=DNS:<YOUR_HOST_NAME>")
+
+# Create a k8s secret
+kubectl create secret tls superset-tls --cert=server.crt --key=server.key
+```
+
+This secret name is provided to Charmed Superset and related to the Nginx Ingress Integrator:
+
+```bash
+# Configure the secret name
+juju config superset-k8s tls-secret-name=superset-tls
+
+# Configure the hostname
+juju config superset-k8s external-hostname=<YOUR_HOST_NAME>
+
+# Relate to the Integrator Charm
+juju relate superset-k8s nginx-ingress-integrator
+```
+
+You can now validate your ingress has been created with the tls certificates provided:
+```bash
+# Note the ingress with name like relation_<num>-<hostname>-ingress
+kubectl get ingress
+
+kubectl describe <YOUR_INGRESS_NAME>
+```
+You should see this using the K8s secret provided.
+
+### Enable Google Oauth
+Enabling Google Oauth for Charmed Superset allows users to authenticate using their Google accounts, streamlining login and increasing security.
+
+To enable Google Oauth, you will need a Google project, you can create one [here](https://console.cloud.google.com/projectcreate).
+
+#### Obtain Oauth2 credentials
+If you do not already have Oauth2 credentials set up then follow the steps below:
+1. Navigate to https://console.cloud.google.com/apis/credentials 
+2. Click `+ Create Credentials` 
+3. Select `Oauth client ID`
+4. Select application type (Web application)
+5. Name the application
+6. Add an Authorized redirect URI (`https://<host>:8088/oauth-authorized/google`)
+7. Create and download your client ID and client secret
+
+### Apply oauth configuration to Superset charm
+Create a file `oauth.yaml` using the Oauth2 credentials you obtained from Google, following the example below and replacing the values.
+```yaml
+superset-k8s:
+  google-client-id: "client_id"
+  google-client-secret: "client_secret"
+  oauth-domain: "companydomain.com"
+  oauth-admin-email: "user@companydomain.com"
+```
+This file can now be applied to Charmed Superset with:
+
+```bash
+juju config superset-k8s --file=path/to/oauth.yaml
+```
+
+### Configure the self-registraton role
+By default, with Google Oauth, a Superset user account is automatically created following successful authentication. This user is provided the least privileged role of `Public`, this role can then be elevated in the UI or via API by an `Admin` user. 
+
+To change the role that is applied on self-registration, simply pass the role via the configuration parameter `self-registration-role`. Superset's standard roles and their associated permissions can be found [here](https://github.com/apache/superset/blob/master/RESOURCES/STANDARD_ROLES.md).
+
+```bash
+juju config superset-k8s self-registration-role=Alpha
+```
+
+### Configuring HTML Sanitization (enabled by default)
```

Comment:
```suggestion
### Configuring HTML Sanitization 
```

Reply:
If the "by default" is important enough, you can add a note box below the first paragraph, saying "Charmed Superset enables by default the sanitization of HTML content" or so